Mathlib/Probability/Distributions/Gaussian.lean

```diff
@@ -347,6 +348,52 @@ lemma gaussianReal_mul_const {X : Ω → ℝ} (hX : Measure.map X ℙ = gaussian
 
 end Transformations
 
+open Measurable Real
+
+variable {Ω : Type*} {mΩ : MeasurableSpace Ω} {p : Measure Ω} {μ : ℝ} {v : ℝ≥0} {X : Ω → ℝ}
+
+lemma mgf_dirac (hX : p.map X = .dirac μ) (t : ℝ) : mgf X p t = exp (μ * t) := by
+  rw [← mgf_id_map, mgf, hX, integral_dirac, mul_comm]
+  · rfl
+  · apply AEMeasurable.of_map_ne_zero
+    rw [hX]
+    exact IsProbabilityMeasure.ne_zero (.dirac μ)
+
+theorem mgf_gaussianReal (hX : p.map X = gaussianReal μ v) (t : ℝ) :
+    mgf X p t = exp (μ * t + v * t ^ 2 / 2) := by
+  by_cases hv : v = 0
+  · simp only [gaussianReal, hv, ↓reduceIte] at hX
+    rw [mgf_dirac hX, hv]
+    simp only [NNReal.coe_zero, zero_mul, zero_div, add_zero]
```

Comment:
Please avoid squeezing terminal simps
```suggestion
    simp [mgf_dirac hX, hv]
```

Reply:
why? I thought `simp only` is more efficient than `simp`. Why would `simp` ever be preferred over `simp only`?

Reply:
Because it's more versatile. Every lemma you mention by name is something that could break due to either:
* the lemma being renamed
* the lemma changing statement

Reply:
I see. What about efficiency though? Isn't `simp` slower than `simp only`?

Reply:
Actually, not always. In some cases, a simp call is so fast that squeezing it (turning it into a `simp only` with a bunch of lemmas) is slower because of the time it takes to parse the list of lemmas.

Here, it certainly isn't substantially slowing anything down